master/internal/api_command.go

```diff
@@ -100,41 +95,22 @@ func (a *apiServer) getCommandLaunchParams(ctx context.Context, req *protoComman
 	if req.MustZeroSlot {
 		resources.Slots = 0
 	}
-	poolName, err := a.m.rm.ResolveResourcePool(
-		resources.ResourcePool, int(cmdSpec.Metadata.WorkspaceID), resources.Slots)
-	if err != nil {
-		return nil, nil, status.Errorf(codes.InvalidArgument, err.Error())
-	}
-	if err = a.m.rm.ValidateResources(poolName, resources.Slots, true); err != nil {
-		return nil, nil, fmt.Errorf("validating resources: %v", err)
-	}
 
-	launchWarnings, err := a.m.rm.ValidateResourcePoolAvailability(
-		&sproto.ValidateResourcePoolAvailabilityRequest{
-			Name:  poolName,
-			Slots: resources.Slots,
-		},
+	poolName, launchWarnings, err := a.m.ResolveResources(
+		resources.ResourcePool,
+		resources.Slots,
+		int(cmdSpec.Metadata.WorkspaceID),
+		true,
 	)
 	if err != nil {
-		return nil, launchWarnings, fmt.Errorf("checking resource availability: %v", err.Error())
-	}
-	if a.m.config.ResourceManager.AgentRM != nil &&
-		a.m.config.LaunchError &&
-		len(launchWarnings) > 0 {
-		return nil, nil, errors.New("slots requested exceeds cluster capacity")
+		return nil, nil, err
```

Comment:
how is the `launchWarnings` from above being used? should this be
```suggestion
		return nil, launchWarnings, err
```